int32 regenerate_env
int32 select_env
---
